### PR TITLE
ISPN-9161 Incorrect description StateTransferManager attributes

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.redhat.com/browse/ISPN-9161

### CODE REVIEW

File: core/src/main/java/org/infinispan/statetransfer/StateTransferManagerImpl.java [hunk 1]
```diff
@@ -238,16 +238,16 @@ public void waitForInitialStateTransferToComplete() {
...
                           rpcManager.getAddress());
-            boolean success = initialStateTransferComplete.await(configuration.clustering().stateTransfer().timeout(), TimeUnit.MILLISECONDS);
-            if (!success) {
-               throw new CacheException(String.format("Initial state transfer timed out for cache %s on %s",
-                                                      cacheName, rpcManager.getAddress()));
-            }
+            initialStateTransferComplete.get(configuration.clustering().stateTransfer().timeout(),
+                                             TimeUnit.MILLISECONDS);
+         } catch (TimeoutException e) {
+            throw new CacheException(String.format("Initial state transfer timed out for cache %s on %s",
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/statetransfer/StateTransferManagerImpl.java:249]: move the message to `Log`?

File: core/src/main/java/org/infinispan/statetransfer/StateTransferManagerImpl.java [hunk 2]
```diff
@@ -90,7 +90,7 @@
    @Inject IracManager iracManager;
    @Inject IracVersionGenerator iracVersionGenerator;
 
-   private final CountDownLatch initialStateTransferComplete = new CountDownLatch(1);
+   private final CompletableFuture<Void> initialStateTransferComplete = new CompletableFuture<>();
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/statetransfer/StateTransferManagerImpl.java:93]: hmm, isn't a `CompletableFuture` kind of overkill?

Comment (danberindei) [reply core/src/main/java/org/infinispan/statetransfer/StateTransferManagerImpl.java:93]: The `CountDownLatch.getCount()` javadoc says it should only used for testing, so I wanted to get rid of it, and `CompletableFuture.isDone()` sounded cleaner than `await(0, MILLISECONDS)`

Comment (pruivo) [reply core/src/main/java/org/infinispan/statetransfer/StateTransferManagerImpl.java:93]: sometimes I have the feeling we are "abusing" the `CompletableFuture` when a simples `Semaphore`, `CountDownLatch`, `boolean` is enough :rofl:

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
merged! thanks @danberindei !

**Pull request merged**